### DISCUSSION TITLE
Understanding the rbf cost code

### DISCUSSION THREAD

**Discussion by mhbl3** [deepcharles/ruptures#291] [Q&A] [answered]
Hello,

I hope this question is not because I am rusty in linear algebra but I have been struggling to understand how $c(y_{I}) = \sum_{t \in I}||\Phi(y_{t})- \overline{\mu}_{t}| |^2_{H}$ became this in the code:

```python 
sub_gram = self.gram[start:end, start:end]
val = np.diagonal(sub_gram).sum()
val -= sub_gram.sum() / (end - start)
```

Thank you!

**Comment by deepcharles** [comment, marked as answer]
Hi, sorry for the late reply.
You can refer to Remark 2 in [this article](https://charles.doffy.net/files/sp-review-2020.pdf) and note that the Gram matrix contains the values of $k(y_{i}, y_{j})$ at the position $(i,j)$. 

hope this is clear enough.
